tests/unit/test_jenkins.py

```diff
@@ -322,38 +322,42 @@ def test__install_plugins(
 def test__configure_proxy_fail(
     harness_container: HarnessWithContainer,
     proxy_config: state.ProxyConfig,
+    mock_client: unittest.mock.MagicMock,
     raise_exception: typing.Callable,
+    monkeypatch: pytest.MonkeyPatch,
 ):
     """
     arrange: given a test proxy config and a monkeypatched jenkins client that raises an exception.
     act: when _configure_proxy is called.
     assert: JenkinsProxyError is raised.
     """
-    mock_client = unittest.mock.MagicMock(spec=jenkinsapi.jenkins.Jenkins)
     mock_client.run_groovy_script = lambda *_args, **_kwargs: raise_exception(
         exception=jenkinsapi.custom_exceptions.JenkinsAPIException
     )
+    monkeypatch.setattr(jenkins, "_get_client", lambda *_args, **_kwargs: mock_client)
 
     with pytest.raises(jenkins.JenkinsProxyError) as exc:
-        jenkins._configure_proxy(harness_container.container, proxy_config, mock_client)
+        jenkins._configure_proxy(harness_container.container, proxy_config)
 
     assert exc.value.args[0] == "Proxy configuration failed."
 
 
 def test__configure_proxy_partial(
     harness_container: HarnessWithContainer,
     partial_proxy_config: state.ProxyConfig,
+    monkeypatch: pytest.MonkeyPatch,
+    mock_client: unittest.mock.MagicMock,
 ):
     """
     arrange: given a test partial proxy config and a mock jenkins client.
     act: when _configure_proxy is called.
     assert: mock client is called with correct proxy arguments.
     """
-    mock_client = unittest.mock.MagicMock(spec=jenkinsapi.jenkins.Jenkins)
     mock_run_groovy_script = unittest.mock.MagicMock()
     mock_client.run_groovy_script = mock_run_groovy_script
+    monkeypatch.setattr(jenkins, "_get_client", lambda *_args, **_kwargs: mock_client)
```

Comment:
Shouldn't this use the client fixture?

Reply:
As in, can't the monkeypatch be done in the mock_client fixture?

Reply:
Saved a lot of lines of code, thanks for catching that!